apollo-router/src/plugins/enhanced_client_awareness/mod.rs

```diff
@@ -0,0 +1,87 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt;
+
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::plugins::telemetry::CLIENT_LIBRARY_NAME;
+use crate::plugins::telemetry::CLIENT_LIBRARY_VERSION;
+use crate::services::supergraph;
+
+const CLIENT_LIBRARY_KEY: &str = "clientLibrary";
+const CLIENT_LIBRARY_NAME_KEY: &str = "name";
+const CLIENT_LIBRARY_VERSION_KEY: &str = "version";
+
+#[derive(Debug, Deserialize, JsonSchema)]
+struct Config {
+    #[serde(default = "default_enable_client_library_metrics")]
+    enable_client_library_metrics: bool,
+}
+
+fn default_enable_client_library_metrics() -> bool {
+    true
+}
+
+struct EnhancedClientAwareness {
+    enabled: bool,
+}
+
+#[async_trait::async_trait]
+impl Plugin for EnhancedClientAwareness {
+    type Config = Config;
+
+    // This is invoked once after the router starts and compiled-in
+    // plugins are registered
+    async fn new(init: PluginInit<Self::Config>) -> Result<Self, BoxError> {
+        Ok(EnhancedClientAwareness {
+            enabled: init.config.enable_client_library_metrics,
```

Comment:
I don't actually want this plugin to be configurable by end users. IMO it should always be executed and the configuration is placed at the client where the library metadata can be sent or disabled.

`Config` seems to be a required part of a plugin though?

Reply:
You should be able to use the empty tuple `()` for the config.

Reply:
After discussing a bit, it looks like `()` actually doesn't work right now. https://github.com/apollographql/router/blob/7041a1a565e79aff6298b890dc7b5bd61ca4dbb9/apollo-router/src/plugins/license_enforcement/mod.rs is another always-on plugin that has no user config: it uses an empty struct.